Add RTL language and localization support to image controls

yoelhawa opened the pull request:
#### bc47e75992068516a93f529867db051f59073526
<pre>
Add RTL language and localization support to image controls
<a href="https://bugs.webkit.org/show_bug.cgi?id=299915">https://bugs.webkit.org/show_bug.cgi?id=299915</a>
&lt;<a href="https://rdar.apple.com/161690817">rdar://161690817</a>&gt;

Reviewed by Abrar Rahman Protyasha.

Use localized string for image controls label. Also detect system RTL
language usage and adjust layout as appropriate.

* Source/WebCore/html/shadow/SpatialImageControls.cpp:
(WebCore::SpatialImageControls::ensureSpatialControls):
* Source/WebCore/html/shadow/spatialImageControls.css:
(div#label):
(span#spatial-glyph):
(span#pano-glyph):
Mirror layout iff system is in RTL language.

* Source/WebCore/en.lproj/Localizable.strings:
* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::imageControlsLabelSpatial):
(WebCore::imageControlsLabelPanorama):
* Source/WebCore/platform/LocalizedStrings.h:
Make label text localizable strings.

* LayoutTests/TestExpectations:
* LayoutTests/fast/images/spatial-image-controls-expected.html:
* LayoutTests/fast/images/spatial-image-controls-rtl-expected.html:
* LayoutTests/fast/images/spatial-image-controls-rtl.html:
* LayoutTests/platform/visionos/TestExpectations:
Add test coverage for RTL support.

Canonical link: <a href="https://commits.webkit.org/300966@main">https://commits.webkit.org/300966@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1e33bdf88a8e44269896c22731bfbd88708cf2cc

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124479 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44159 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/34894 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131322 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/76470 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/126356 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/44884 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/52745 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/94696 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/76470 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127433 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/35776 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111331 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75274 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/34713 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/29491 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/74802 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/105528 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/29718 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/133986 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51360 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39185 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/103176 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/51766 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107553 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/102961 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26211 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48324 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/26583 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/48303 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51230 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57017 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/50641 "Built successfully") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/54001 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52321 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebCore/html/shadow/SpatialImageControls.cpp:
```diff
@@ -130,6 +131,9 @@ void ensureSpatialControls(HTMLImageElement& imageElement)
         if (!isSpatialImage && !isPanoramicImage)
             return;
 
+        Page* page = element->document().page();
```
```suggestion
        RefPtr page = element->document().page();
```

etiennesegonzac commented on Source/WebCore/html/shadow/spatialImageControls.css:
```diff
@@ -112,8 +112,8 @@ span#pano-glyph {
-    margin-right: 5px;
-    margin-left: 10px;
+    margin-inline-end: 5px;
+    margin-inline-start: 10px;
```
since we already have reftests we might want to add one for RTL :)

yoelhawa replied:
Good call!

Commit:
Add RTL language and localization support to image controls
https://bugs.webkit.org/show_bug.cgi?id=299915
<rdar://161690817>

Reviewed by Abrar Rahman Protyasha.

Use localized string for image controls label. Also detect system RTL
language usage and adjust layout as appropriate.

* Source/WebCore/html/shadow/SpatialImageControls.cpp:
(WebCore::SpatialImageControls::ensureSpatialControls):
* Source/WebCore/html/shadow/spatialImageControls.css:
(div#label):
(span#spatial-glyph):
(span#pano-glyph):
Mirror layout iff system is in RTL language.

* Source/WebCore/en.lproj/Localizable.strings:
* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::imageControlsLabelSpatial):
(WebCore::imageControlsLabelPanorama):
* Source/WebCore/platform/LocalizedStrings.h:
Make label text localizable strings.

* LayoutTests/TestExpectations:
* LayoutTests/fast/images/spatial-image-controls-expected.html:
* LayoutTests/fast/images/spatial-image-controls-rtl-expected.html:
* LayoutTests/fast/images/spatial-image-controls-rtl.html:
* LayoutTests/platform/visionos/TestExpectations:
Add test coverage for RTL support.

Canonical link: https://commits.webkit.org/300966@main
